.gitlab-ci.yml

```diff
@@ -0,0 +1,141 @@
+stages:
+  - code quality
+  - dependency scanning
+  - build
+  - test
+  - deploy
+
+
+# Code Quality section
+Check code standards with PHP_CodeSniffer:
+  image: sumocoders/framework-php72:latest
+  before_script:
+    - curl -sS https://getcomposer.org/installer | php
+    - php composer.phar install --no-scripts --quiet --ignore-platform-reqs
+  script:
+    - vendor/bin/phpcs --report-full --report-junit=phpcs-report.xml
+  artifacts:
+    expire_in: 1 week
+    reports:
+      junit: phpcs-report.xml
+  stage: code quality
+  tags:
+    - docker
+  allow_failure: true
+
+Check for bugs with PHPStan:
+  image: sumocoders/framework-php72:latest
+  before_script:
+    - curl -sS https://getcomposer.org/installer | php
+    - php composer.phar install --no-scripts --quiet --ignore-platform-reqs
+    - bin/console cache:warmup --env=dev
+  script:
+    - vendor/bin/phpstan analyse --memory-limit=512M
+    - vendor/bin/phpstan analyse --memory-limit=512M --error-format=junit --no-progress > phpstan-report.xml
+  artifacts:
+    expire_in: 1 week
+    reports:
+      junit: phpstan-report.xml
+  stage: code quality
+  tags:
+    - docker
+  allow_failure: true
+
+
+## Dependency Scanning section
+Check NPM dependencies for vulnerabilities:
+  image: sumocoders/cli-tools-php72:latest
+  before_script:
+    - curl -sS https://getcomposer.org/installer | php
+    - php composer.phar install --no-scripts --quiet --ignore-platform-reqs
+  script:
+    - vendor/bin/convert-to-junit-xml convert:npm-audit "$(npm audit --json)" > npm-audit-report.xml
+  artifacts:
+    expire_in: 1 week
+    reports:
+      junit: npm-audit-report.xml
+  stage: dependency scanning
+  tags:
+    - docker
+  allow_failure: true
+
+Check Composer dependencies for vulnerabilities:
+  image: sumocoders/cli-tools-php72:latest
+  before_script:
+    - curl -sS https://getcomposer.org/installer | php
+    - php composer.phar install --no-scripts --quiet --ignore-platform-reqs
+  script:
+    - vendor/bin/convert-to-junit-xml convert:sensiolabs-security-check "$(vendor/bin/security-checker security:check --format=json)" > security-checker-report.xml
+  artifacts:
+    expire_in: 1 week
+    reports:
+      junit: security-checker-report.xml
+  stage: dependency scanning
+  tags:
+    - docker
+  allow_failure: true
+
+
+## Build section
```

Comment:
Why is this commented?

Reply:
It was by mistake, I copied it from another project assuming everything was enabled.

Reply:
It's enabled again.